x-pack/plugins/infra/public/components/asset_details/hooks/use_page_header.tsx

```diff
@@ -107,10 +108,27 @@ const useRightSideItems = (links?: LinkOptions[]) => {
   return { rightSideItems };
 };
 
+const useFeatureFlagTabs = () => {
+  const { featureFlags } = usePluginConfig();
+  const featureFlagTabs: Partial<Record<ContentTabIds, { isEnabled: () => boolean }>> = useMemo(
+    () => ({
+      [ContentTabIds.OSQUERY]: {
+        isEnabled: () => featureFlags.osqueryEnabled,
+      },
+    }),
+    [featureFlags.osqueryEnabled]
+  );
+
+  return {
+    featureFlagTabs,
+  };
+};
+
```

Comment:
A few suggestions here: 
* The result of this hook is a bit cumbersome to use because it has only a partial map of the tab IDs. Having the full map ` { [key in ContentTabIds]: { isEnabled: boolean }}` as a result would be simpler to use.
* Seems like there is no need for `isEnabled` to be a function

Reply:
Not all tabs are controlled by feature flags, that's why I wanted to be very specific here. Agree on the `isEnabled` function.